test/test_analyse.ml

```diff
@@ -0,0 +1,114 @@
+open Lwt.Infix
+
+module Analysis = struct
+  include Ocaml_ci.Analyse.Analysis
+
+  let set_equality = Alcotest.(equal (slist string String.compare))
+
+  (* Make the [t] type concrete from the observable fields for easier testing *)
+  type t = {
+    opam_files : string list; [@equal set_equality]
+    is_duniverse : bool;
+    ocamlformat_version : string option;
+  }
+  [@@deriving eq, yojson]
+
+  let of_dir ~job d =
+    of_dir ~job d
+    |> Lwt_result.map (fun t ->
+           {
+             opam_files = opam_files t;
+             is_duniverse = is_duniverse t;
+             ocamlformat_version = ocamlformat_version t;
+           })
```

Comment:
This may be out of sync in the future. What do you think of exposing `Analyse.to_yojson` and comparing the json output ?

Reply:
'out-of-sync' is a matter of perspective :slightly_smiling_face: From my perspective, we should test `Analysis` via the properties that it exposes in its signature; the fact that these are 'in sync' at the moment is more a coincidence than anything else. If a new field is added to `Analysis.t`, then I don't think that's a reason artificially break existing tests of other aspects of the API.

From a philosophical perspective, testing equality via equality of pretty-printed output feels wrong to me: `Analysis.t` need not expose its entire public API via its pretty-printer.

Reply:
(if you wanted to change the public API of `Analysis` to make `t` concrete, then I might agree with you...)

Reply:
I agree. Breaking all the tests every time is a bit too much. I was suggesting comparing the Yojson representation but that's still not nice, I agree.

Reply:
Comparing the Yojson representations would break every time, surely? Unless you were suggesting asserting that one Yojson object was a non-strict subset of the other, rather than pure equality.